baybe/utils/chemistry.py

```diff
@@ -72,160 +69,73 @@ def name_to_smiles(name: str) -> str:
-
-    Args:
-        smiles_list: List of SMILES strings.
-        prefix: Name prefix for each descriptor (e.g., nBase --> <prefix>_nBase).
-        dropna: If ``True``, drops columns that contain NaNs.
+    return fingerprint_encoder.transform([smiles_str])
 
-    Returns:
-        Dataframe containing overlapping RDKit descriptors for each SMILES string.
-    """
-    mols = smiles_to_molecules(smiles_list)
-
-    res = []
-    for mol in mols:
-        desc = {
-            prefix + "RDKIT_" + dname: DTypeFloatNumpy(func(mol))
-            for dname, func in Chem.Descriptors.descList
-        }
-        res.append(desc)
-
-    df = pd.DataFrame(res)
-    if dropna:
-        df = df.dropna(axis=1)
-
-    return df
 
-
-def smiles_to_fp_features(
+def smiles_to_fingerprint_features(
     smiles_list: list[str],
+    fingerprint_name: str,
     prefix: str = "",
-    dtype: type[int] | type[float] = int,
-    radius: int = 4,
-    n_bits: int = 1024,
+    kwargs_conformer: dict | None = None,
+    kwargs_fingerprint: dict | None = None,
 ) -> pd.DataFrame:
-    """Compute standard Morgan molecule fingerprints for a list of SMILES strings.
+    """Compute molecular fingerprints for a list of SMILES strings.
 
     Args:
         smiles_list: List of SMILES strings.
+        fingerprint_name: Name of Fingerprint class used to
+            transform smiles to fingerprints
         prefix: Name prefix for each descriptor (e.g., nBase --> <prefix>_nBase).
-        dtype: Specifies whether fingerprints will have int or float data type.
-        radius: Radius for the Morgan fingerprint.
-        n_bits:Number of bits for the Morgan fingerprint.
+        kwargs_conformer: kwargs for ConformerGenerator
+        kwargs_fingerprint: kwargs for ConformerGenerator
 
     Returns:
-        Dataframe containing Morgan fingerprints for each SMILES string.
+        Dataframe containing fingerprints for each SMILES string.
     """
-    mols = smiles_to_molecules(smiles_list)
-
-    res = []
-    for mol in mols:
-        RDLogger.logger().setLevel(RDLogger.CRITICAL)
-
-        fingerp = GetMorganFingerprintAsBitVect(mol, radius, nBits=n_bits).ToBitString()
-        fingerp = map(int, fingerp)
-        fpvec = np.array(list(fingerp))
-        res.append(
-            {prefix + "FP_" + f"{k + 1}": dtype(bit) for k, bit in enumerate(fpvec)}
+    kwargs_fingerprint = {} if kwargs_fingerprint is None else kwargs_fingerprint
+    fingerprint_encoder = getattr(skfp_fingerprints, fingerprint_name)(
+        **kwargs_fingerprint
+    )
+
+    if fingerprint_encoder.requires_conformers or (
+        # TODO removed once fixed in skfp package, see
+        #  https://github.com/scikit-fingerprints/scikit-fingerprints/issues/239
+        fingerprint_name in ["USRCATFingerprint", "USRFingerprint"]
+    ):
+        kwargs_conformer = {} if kwargs_conformer is None else kwargs_conformer
+        smiles_list = ConformerGenerator(**kwargs_conformer).transform(
```

Comment:
is the output of `ConformerGenerator` reallya SMILES? if not perhaps some renamings / restruturing to make that clear is better